lib/matplotlib/axes/_axes.py

```diff
@@ -830,10 +831,10 @@ def axhline(self, y=0, xmin=0, xmax=1, **kwargs):
 
             >>> axhline(y=.5, xmin=0.25, xmax=0.75)
         """
+        self._check_no_units([xmin, xmax], ['xmin', 'xmax'])
         if "transform" in kwargs:
-            raise ValueError(
-                "'transform' is not allowed as a kwarg;"
-                + "axhline generates its own transform.")
+            raise ValueError("'transform' is not allowed as a kwarg; "
```

Comment:
This should be spelled out:
```suggestion
            raise ValueError("'transform' is not allowed as a keyword argument; "
```